src/gui/org/deidentifier/arx/gui/view/impl/menu/DialogOrderSelection.java

```diff
@@ -193,34 +195,56 @@ private boolean isValidDataType(DataType<?> type, String[] values) {
     }
     
     /**
-     * Loads the array from a file.
+     * Loads the array from a file. If the file contains more or additional
+     * values (lines) than present in the attribute's domain, the loading is
+     * aborted and an <code>IllegalStateException</code> is thrown.
      *
      * @param file
-     * @param charset TODO
+     * @param charset
+     *            TODO
      * @return
+     * @throws IllegalStateException
+     *             The file contains more or additional values (lines) than
+     *             present in the attribute's domain
      */
-    private String[] loadArray(String file, Charset charset) {
-        ArrayList<String> list = new ArrayList<String>();
+    private java.util.List<String> loadFile(String file,
+                                            Charset charset) throws IllegalStateException {
+        java.util.List<String> list = new ArrayList<String>();
         BufferedReader reader = null;
         try {
             reader = new BufferedReader(new InputStreamReader(new FileInputStream(file), charset));
             String line = reader.readLine();
+            Set<String> _elements = new HashSet<String>();
+            _elements.addAll(Arrays.asList(elements));
+
             while (line != null) {
                 list.add(line);
+                if (list.size() > _elements.size() || !_elements.contains(line)) {
+                    // The file contains more or additional values (lines) than
+                    // present in the attribute's domain
+                    throw new IllegalStateException();

```

Comment:
I think, it would be a better idea to show the error dialog here and then simply return "null", please also implement the check whether list.size() != elements.length also at the end of this message.